src/mappings/mappingHandlers.ts

```diff
@@ -105,9 +105,21 @@ export async function handleToolingEvent(event: SubstrateEvent): Promise<void> {
   const eventId = event.event.method;
   const args = event.event.data.toArray();
 
-  const harvesterLikeArgs = args.map((arg, i) => ({
-    value: serializeLikeHarvester(arg, event.event.meta.args[i].toString(), logFoundType),
-  }));
+  const harvesterLikeArgs = args.map((arg, i) => {
+    let type;
+    const typeName = event.event.meta.fields[i].typeName;
+    if (typeName.isSome) {
+      // for metadata >= v14
+      type = typeName.unwrap().toString();
+    } else {
+      // for metadata < v14
+      type = event.event.meta.args[i].toString();
```

Comment:
I thought the bad values were from old events. Isn't this the same as it was before for metadata < 14?

Reply:
No, it isn't the same after the upgrade. I added the old vs new values here - https://polymathnetwork.slack.com/archives/C026L2FKT25/p1661246441250979?thread_ts=1661203792.561179&cid=C026L2FKT25